src/eos-companion-app-service-managed-cache.c

```diff
@@ -135,43 +86,78 @@ eos_companion_app_service_managed_cache_clear (EosCompanionAppServiceManagedCach
 {
   EosCompanionAppServiceManagedCachePrivate *priv = eos_companion_app_service_managed_cache_get_instance_private (cache);
 
-  threadsafe_registered_cache_clear (priv->registered_cache);
+  g_mutex_lock (&priv->mutex);
+  g_hash_table_remove_all (priv->cache_tree);
+  g_mutex_unlock (&priv->mutex);
 }
 
 /**
- * eos_companion_app_service_managed_cache_get_subcache: (skip)
- * @cache: A #EosCompanionAppServiceManagedCache.
- * @key: The key to lookup the subcache with.
+ * eos_companion_app_service_managed_cache_lock_subcache: (skip)
+ * @cache: An #EosCompanionAppServiceManagedCache.
+ * @key: The key to lock the subcache with.
  * @value_destroy: A #GDestroyNotify for the value type of the subcache, if
  *                 it needs to be created.
  *
- * Get a #ThreadsafeRegisteredCache from a #EosCompanionAppServiceManagedCache.
+ * Get a #GHashTable subcache from a #EosCompanionAppServiceManagedCache. The
+ * subcache is in a "locked" state, meaning that no other thread can read from
+ * or write to this subcache. Once the caller is done with the subcache, it
+ * should return it by using 
```

Comment:
This paragraph is truncated

Reply:
Oops, must have intended to copy-paste the function name and not gotten back to it